server/routes/ratings.js

```diff
@@ -0,0 +1,81 @@
+var express = require('express');
+var router = express.Router();
+var models = require('../models');
+
+/* GET ratings listing. */
+router.get('/', function (req, res) {
+  models.Rating.findAll().then(function (ratings) {
+    var mappedRatings = ratings.map(function (rating) {
+      return ({ value: rating.value,
+                userdId: rating.UserId,
+                snippetId: rating.SnippetId
+      });
+    });
+
+    res.send(mappedRatings);
+  });
+});
+
+/* GET rating by id */
+router.get('/:id', function (req, res) {
+  models.Rating.findById(req.params.id).then( function (rating) {
+    res.send(rating);
+  });
+});
+
+/* UPDATE user's rating for snippet*/
+router.put('/:id', function (req, res) {
+  models.Rating.findById(req.params.id).then( function (rating) {
+    rating.value = req.body.value;
+    rating.save({ validate: false, logging: true}).then(function () {
+      res.status(200).send('ok');
+    }).catch(function () {
+      res.status(422).send('error');
+    });
+  }).catch(function () {
+    res.status(422).send('error');
+  });
+});
+
+/* POST new rating */
+router.post('/', function (req, res) {
+  var user_id = req.user.dataValues.id;
+  models.Rating.findOne({ where : { SnippetId: req.body.SnippetId, UserId: user_id } }).then( function (rating) {
+    if (rating) {
+      var snippetAverage = 0;
+      var sum = 0.0;
+      var index = 0;
+      rating.value = req.body.value;
+      models.Rating.findAll({ where: {SnippetId: req.body.SnippetId} }).then( function (ratings) {
```

Comment:
@dwoznicka I think we should use here builtin SQL functions like ```AVG``` and ```COUNT``` not iterate over an array.